examples/gno.land/r/demo/vrf/vrf.gno

```diff
@@ -0,0 +1,269 @@
+type InvalidByteError byte
+
+func (e InvalidByteError) Error() string {
+	return fmt.Sprintf("encoding/hex: invalid byte: %#U", rune(e))
+}
+
+func EncodeToHexString(src []byte) string {
+	encoded := []byte{}
+	j := 0
+	for _, v := range src {
+		encoded = append(encoded, hextable[v>>4])
+		encoded = append(encoded, hextable[v&0x0f])
+	}
+	return string(encoded)
+}
+
+func DecodeFromHexString(src []byte) ([]byte, error) {
+	dst := []byte{}
+	i, j := 0, 1
+	for ; j < len(src); j += 2 {
+		p := src[j-1]
+		q := src[j]
+
+		a := reverseHexTable[p]
+		b := reverseHexTable[q]
+		if a > 0x0f {
+			return dst, InvalidByteError(p)
+		}
+		if b > 0x0f {
+			return dst, InvalidByteError(q)
+		}
+		dst = append(dst, (a<<4)|b)
+	}
+	if len(src)%2 == 1 {
+		// Check for invalid char before reporting bad length,
+		// since the invalid char (if present) is an earlier problem.
+		if reverseHexTable[src[j-1]] > 0x0f {
+			return dst, InvalidByteError(src[j-1])
+		}
+		return dst, ErrLength
+	}
+	return dst, nil
+}
+
+func RenderConfig() string {
+	quotedFeeders := []string{}
+	for _, feeder := range config.feeders {
+		quotedFeeders = append(quotedFeeders, `"`+feeder+`"`)
+	}
+	feedersText := strings.Join(quotedFeeders, ", ")
+	return fmt.Sprintf(`{
+		"vrfAdmin": "%s",
+		"feeders": [%s]
+}`, config.vrfAdmin, feedersText)
+}
+
+func GetRequests(startAfter, limit uint64) []Request {
+	max := uint64(len(requests))
+	if startAfter+limit < max {
+		max = startAfter + limit
+	}
+	return requests[startAfter:max]
+}
+
+func RenderRequest(requestId uint64) string {
+	if requestId >= uint64(len(requests)) {
+		panic("invalid request id")
+	}
+
+	r := requests[requestId]
+	quotedFulfillers := []string{}
+	for _, fulfiller := range r.fulfillers {
+		quotedFulfillers = append(quotedFulfillers, `"`+fulfiller+`"`)
+	}
+	fulfillers := strings.Join(quotedFulfillers, ", ")
+	randomWords := EncodeToHexString(r.randomWords)
+
+	return fmt.Sprintf(`{
+		"id": %d,
+		"requesterAddress": "%s",
+		"requesterRealm": "%s",
+		"requiredFeedersCount": %d,
+		"fulfilledCount": %d,
+		"randomWords": "%s",
+		"fulfillers": [%s]
+}`, r.id, r.requesterAddress, r.requesterRealm, r.requiredFeedersCount, r.fulfilledCount, randomWords, fulfillers)
+}
+
+func RenderRequests(startAfter uint64, limit uint64) string {
+	paginatedRequests := GetRequests(startAfter, limit)
+	rendered := "["
+	for index, req := range paginatedRequests {
+		rendered += RenderRequest(req.id)
+		if index != len(paginatedRequests)-1 {
+			rendered += ",\n"
+		}
+	}
+	rendered += "]"
+	return rendered
+}
+
+func SetVRFAdmin(vrfAdmin string) {
+	if config.vrfAdmin == "" {
+		config.vrfAdmin = vrfAdmin
+		return
+	}
+	caller := std.GetOrigCaller()
+	if config.vrfAdmin != caller.String() {
+		panic("not allowed to update vrfAdmin")
+	}
+
+	config.vrfAdmin = vrfAdmin
+}
+
+func SetFeeders(feedersText string) {
+	feeders := strings.Split(feedersText, ",")
+	caller := std.GetOrigCaller()
```

Comment:
Maybe parse `feedersText` after checking this; no use in parsing if the caller isn't the admin.

Reply:
Here's the fix for the comment
https://github.com/TERITORI/gno/pull/11/commits/8db3d0d30410c61421b668e3d9cfb9d53d0762e0